src/utilities.jl

```diff
@@ -25,7 +25,7 @@ end
 
 Return the average of `a` and `b`
 """
-average(a::T, b::T) where {T<:AbstractFloat} = (a + b) / 2
+average(a::T, b::T) where {T<:AbstractFloat} = (a + b) / T(2)
```

Comment:
Did you run into trouble with this somewhere?  As an integer `2` should be promoted to whatever floating point type `(a+b)` is.

Reply:
I thought there would be an issue is with `Float32` in that the literal `2` is an `Int` which will get promoted to `Float64` on a 64-bit system.  But I was wrong
```julia
julia> (1.0f0 + 2.0f0) / 2
1.5f0
```
so can go back to the original expression.

Reply:
I thought I would leave that in as it doesn't seem to be harmful in any way and it could be protection against unforeseen circumstances.